### DISCUSSION TITLE
Integrating in the NextJs

### DISCUSSION THREAD

**Discussion by mudasirpandith** [excalidraw/excalidraw#7472] [General]
I added the editor in my Nextjs app. Now I want to have a function which will give the data from the editor.

**Comment by othneildrew** [comment]
@mudasirpandith The docs as a list of all the available props that can be passed to the Excalidraw component.
Here is the list: https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/
https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/excalidraw-api

Here is some pseudo code as an example:

```tsx
export default function App() {
  const [excalidrawAPI, setExcalidrawAPI] = useState(null);
  const getData = () => api.getAppState(); // don't forget to call it
  return <Excalidraw excalidrawAPI={{(api)=> setExcalidrawAPI(api)}} />;
}
```

You can also pass the `onChange` prop to the component (consider throttling this)
https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/#onchange
